mava/components/jax/building/environments.py

```diff
@@ -137,3 +138,68 @@ def on_building_executor_environment_loop(self, builder: SystemBuilder) -> None:
                 executor_environment_loop
             )
         builder.store.system_executor = executor_environment_loop
+
+
+@dataclass
+class MonitorExecutorEnvironmentLoopConfig(ExecutorEnvironmentLoopConfig):
+    filename: str = "agents"
+    label: str = "parallel_environment_loop"
+    record_every: int = 1000
+    fps: int = 15
+    counter_str: str = "evaluator_episodes"
+    format: str = "video"
+    figsize: Union[float, Tuple[int, int]] = (360, 640)
+
+
+class MonitorExecutorEnvironmentLoop(ExecutorEnvironmentLoop):
+    def __init__(
+        self,
+        config: MonitorExecutorEnvironmentLoopConfig = MonitorExecutorEnvironmentLoopConfig(),  # noqa
+    ):
+        super().__init__(config=config)
+        self.config = config
+
+    def on_building_executor_environment_loop(self, builder: SystemBuilder) -> None:
+        """
+        Builds a `ParallelEnvironmentLoop` on all executors and a
+        `MonitorParallelEnvironmentLoop` on the executor and stores it
+         in the `builder.store.system_executor`
+
+        Args:
+            builder : the system builder
+        """
+        if builder.store.is_evaluator:
```

Comment:
this if statement is not ideal but it is how we do things elsewhere when we only want this on the evaluator (previously we had eval_loop_fn)